engine/sql/catalyst/src/main/scala/org/apache/spark/sql/spatial/MBR.scala

```diff
@@ -96,7 +96,23 @@ case class MBR(low: Point, high: Point) extends Shape {
     Math.sqrt(ans)
   }
 
-  override def toString: String = "MBR(" + low.toString + "," + high.toString + ")"
+  def intersectRadio(other: MBR) : Double = {
+    require(low.coord.length == other.low.coord.length)
+
+    if (!intersects(other)) {
```

Comment:
Your logic here is still not optimal. You should write something like this:
- Only need to calculate `intersect_low` and `intersect_high`.
- If there is a coordinate such that `intersect_low(i) > intersect_high(i)`, then `return 0.0`.
- Else calculate the area and estimate selectivity.

I think have an `area` function for MBR is a good idea. But do not use it calculate intersect area since it will cause another creation of MBR object. DO use it on current MBR.